sqldelight-gradle-plugin/src/main/kotlin/com/squareup/sqldelight/gradle/SqlDelightDatabase.kt

```diff
@@ -148,6 +145,15 @@ class SqlDelightDatabase(
         it.verifyMigrations = verifyMigrations
       }
 
+      // Add the source dependency on the generated code.
+      source.sourceDirectorySet.srcDir(
+        // Use a Provider generated from the task to carry task dependencies
+        // See https://github.com/cashapp/sqldelight/issues/2119
+        task.map {
```

Comment:
With this are we able to remove the explicit dependency below on line 162?

Reply:
Maybe? I'd expect this to work for Kotlin/Multiplatform project but I'm not 100% sure about Android projects, different versions of AGP, etc... 
I'll try making `registerTaskDependency` a no-op for Kotlin projects for now.

Reply:
Alright, I ended up removing all `registerTaskDependency` (including for Android) and everything is still green so 👍. The only thing I can think of right now is that this won't work for older "pre-Providers" versions of Gradle (not sure which ones, maybe < 6). I guess that's ok now?